pkg/kinflate/resource/appresource.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	kutil "k8s.io/kubectl/pkg/kinflate/util"
+	"k8s.io/kubectl/pkg/loader"
+)
+
+func ResourcesFromPath(path string, loader loader.Loader) ([]*Resource, error) {
```

Comment:
Name is inconsistent with https://github.com/kubernetes/kubectl/blob/master/pkg/kinflate/resource/configmap.go#L29 :-/

Reply:
Yes, this is because that the function here has a different behavior. It returns a slice of Resource. While in configmap.go, it returns one Resource.

Reply:
Yeah, this doesn't show up nicely in the godoc either because of that. We might want to have "Resources" on top of resources (which would help checking for conflicts on top of that). And then we could have consistent creation of resources (even for configmap we could return a list of 1 resource, but then it'd be consistent). I don't know. Too early to waste time on that :-)